### PR TITLE
workflows: add summary of Twister CI tests

### PR DESCRIPTION
**Pull request by szczys**
Collect the suite report from each Twister run and use it to render a pass/fail grid in the checks tab of the PR.

Summaries of the integration tests will be added in an upcoming PR.

Depends on: [szczys/mask-workflow-secrets](https://github.com/golioth/golioth-firmware-sdk/pull/597)
Resolves https://github.com/golioth/firmware-issue-tracker/issues/662

### CODE REVIEW

File: .github/workflows/hil_sample_zephyr.yml [hunk 1]
```diff
@@ -219,6 +220,19 @@ jobs:
             twister-out/*.xml
             twister-out/*.json
 
+      - name: Prepary report summary
```
Comment (szczys) [inline .github/workflows/hil_sample_zephyr.yml:223]:
```suggestion
      - name: Prepare report summary
```

File: .github/workflows/hil_sample_zephyr.yml [hunk 2]
```diff
@@ -219,6 +220,19 @@ jobs:
...
+        if: always() && steps.mask-logs.outcome == 'success'
+        run: |
+          mkdir summary
+          cp twister-out/twister_suite_report.xml summary/twister-${{ inputs.hil_board }}.xml
+
+      - name: Upload report summary
+        uses: actions/upload-artifact@v4
+        if: always() && steps.mask-logs.outcome == 'success'
+        with:
+          name: ci-summary-twister-${{ inputs.hil_board }}
```
Comment (szczys) [inline .github/workflows/hil_sample_zephyr.yml:233]:
```suggestion
          name: ci-summary-samples-zephyr-${{ inputs.hil_board }}
```

Use an artifact title that more closely follows the workflow names

File: .github/workflows/hil_tests.yml [hunk 3]
```diff
@@ -247,6 +247,21 @@ jobs:
             twister-out/*.xml
             twister-out/*.json
 
+      - name: Prepary report summary
```
Comment (szczys) [inline .github/workflows/hil_tests.yml:250]:
```suggestion
      - name: Prepare report summary
```

File: .github/workflows/hil_tests.yml [hunk 4]
```diff
@@ -247,6 +247,21 @@ jobs:
...
+        if: always() && steps.mask-logs.outcome == 'success'
+        run: |
+          mkdir summary
+          cp twister-out/twister_suite_report.xml summary/twister-${{ matrix.artifact_suffix }}.xml
+
+      - name: Upload report summary
+        uses: actions/upload-artifact@v4
+        if: always() && steps.mask-logs.outcome == 'success'
+        with:
+          name: ci-summary-twister-${{ matrix.artifact_suffix }}
```
Comment (szczys) [inline .github/workflows/hil_tests.yml:260]:
```suggestion
          name: ci-summary-samples-zephyr-${{ matrix.artifact_suffix }}
```
Use an artifact title that more closely follows the workflow names

**Comment by szczys** [comment]
This one is ready for review. Use the `checks` tab and select `HIL Test Summary` to see the output: https://github.com/golioth/golioth-firmware-sdk/pull/598/checks?check_run_id=30230523452

This branch will need to be rebased on `szczys/mask-workflow-secrets` before merging but only because I tweaked a commit message over there to re-run tests after today's GitHub Actions outtage cancelled a bunch of our running tests.

File: .github/workflows/hil_tests.yml [hunk 5]
```diff
@@ -328,3 +344,29 @@ jobs:
...
       coap_gateway_url: ${{ inputs.coap_gateway_url }}
     secrets: inherit
+
+  publish_summary:
+    runs-on: ubuntu-latest
+    if: always()
+    needs: [
+              hil_sample_zephyr,
+              hil_sample_zephyr_nsim,
+           ]
```
Comment (mniestroj) [inline .github/workflows/hil_tests.yml:354]:
```suggestion
    needs:
      - hil_sample_zephyr
      - hil_sample_zephyr_nsim
```

**Review by mniestroj — APPROVED** [review]

**Pull request merged**